test/integration/msi_pwsh_int_windows_test.go

```diff
@@ -0,0 +1,56 @@
+package integration
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/ActiveState/cli/internal/environment"
+	"github.com/ActiveState/cli/internal/testhelpers/e2e"
+	"github.com/ActiveState/termtest"
+)
+
+var (
+	installAction   msiExecAction = "install"
+	uninstallAction msiExecAction = "uninstall"
+)
+
+type msiExecAction string
+
+func (a msiExecAction) cmd(msiPath, installPath string) string {
+	msiAct := "/package"
+	if a == uninstallAction {
+		msiAct = "/uninstall"
+	}
+
+	pwshCmdForm := `$proc = Start-Process msiexec.exe -Wait -ArgumentList ` +
+		`'%s %s /quiet /qn /norestart /log %s INSTALLDIR="%s"' -PassThru;` +
+		`echo "exitcode:$($proc.ExitCode):";` + // use to ensure exit code is exactly 0
+		`exit $proc.ExitCode;`
+	pwshCmd := fmt.Sprintf(pwshCmdForm, msiAct, msiPath, a.logFileName(msiPath), installPath)
+	return pwshCmd
+}
+
+func (a msiExecAction) logFileName(msiPath string) string {
+	msiName := filepath.Base(strings.TrimSuffix(msiPath, filepath.Ext(msiPath)))
+	return filepath.Join(environment.GetRootPathUnsafe(), fmt.Sprintf(`%s_%s.log`, msiName, string(a)))
+}
+
+type pwshSession struct {
```

Comment:
@mdrohmann is this really necessary? Seems we're just using powershell to capture the exit code, I have a hard time believing we really need to do that.

Reply:
It is really for the `Start-Process` cmdlet.  The `msiexec.exe` returns right away otherwise, and we have no idea when the installation actually finishes... Well, we could watch the log file, but I think that would be even worse...

The perl cmds do not necessarily have to be called by a powershell.